aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -302,9 +326,36 @@ public void onResult(SignUpResult result) {
 
             @Override
             public void onError(Exception error) {
-                onException.accept(
-                    new AuthException("Confirm sign up failed", error, "See attached exception for more details")
-                );
+                if (error instanceof UserNotFoundException) {
+                    onException.accept(
+                            new AuthException.AWSCognitoAuthException.UserNotFoundException(error.getCause())
```

Comment:
Hmm, any way to shorten these class names?  Looks like you are using `AuthException` and `AWSCognitoAuthException` for two purposes - as a namespace AND as a parent class for the inner class exceptions.  Do you need to do both?  
  
Could you instead name it something like `AWSCognitoAuthUserNotFoundException`, and define it as a top level class?

Reply:
No we don't want to change the top level class - this is one of the main Amplify error tenets that we keep exceptions with the category top level classes so as not to introduce breaking changes.